packages/marshal/src/marshal.js

```diff
@@ -247,52 +247,57 @@ function isPassByCopyArray(val) {
   return true;
 }
 
+/**
+ * Everything having to do with a `dataProto` is a temporary kludge until
+ * we're on the other side of #2018. TODO remove.
+ */
+const dataProto = harden(
+  create(objectPrototype, {
+    [PASS_STYLE]: { value: 'copyRecord' },
+  }),
+);
+
+const isDataProto = proto => {
+  if (!isFrozen(proto)) {
+    return false;
+  }
+  if (getPrototypeOf(proto) !== objectPrototype) {
+    return false;
+  }
+  const {
+    // @ts-ignore
+    [PASS_STYLE]: passStyleDesc,
+    ...rest
```

Comment:
I would think a clean version of this test could be done without an allocation here. It would be nice if a simple test operation was O(1)

Reply:
For throw away code I'm not inclined to block this PR on a performance issue that will quickly go away --- once we're on the other side of #2018 . But we do have similar code for the remotable case so if you have a pattern to suggest, I'd like to hear it. But that code predates this PR so this PR need not wait on improving it.